libautoscoper/src/PSO.cpp

```diff
@@ -38,7 +38,7 @@ void intializeRandom()
 
 float getRandom(float low, float high)
 {
-  return low + (high - low + 1.0f)*(float)rand() / (RAND_MAX + 1.0f);
+  return low + (high - low)*(float)rand() / (float)RAND_MAX;
```

Comment:
```suggestion
  return low + (high - low) * getRandomClamped();
```

Reply:
Adding a table with before/after showing the computed value when `rand()` returns either `0` or `RAND_MAX` would be helpful too.